tests/v2/actions/charts/stackstate.go

```diff
@@ -285,3 +363,31 @@ func newStackstateAgentChartUpgradeAction(p *payloadOpts, stackstateConfigs *obs
 
 	return chartUpgradeAction
 }
+
+// CreateClusterRepo creates a new ClusterRepo resource in the Kubernetes cluster using the provided catalog client.
+// It takes the client, repository name, and repository URL as arguments and returns an error if the operation fails.
+func CreateClusterRepo(client *rancher.Client, catalogClient *catalog.Client, name, url string) error {
+	ctx := context.Background()
+	repo := buildClusterRepo(name, url)
+	_, err := catalogClient.ClusterRepos().Create(ctx, repo, metav1.CreateOptions{})
+
+	client.Session.RegisterCleanupFunc(func() error {
+
+		var propagation = metav1.DeletePropagationForeground
```

Comment:
what is the propogationForeground used for?  can we not delete the repo by repoObject.Name?

Reply:
Honestly this was used in another test's cleanup, so it's possible that it can work the other way. Is there a reason to change it?

Reply:
oh ok. I think its different in each of these charts tests. in one of the charts tests, I noticed we are using the repoObject to delete. not really sure what PropagationForeground is, so was curious if we cannot just use the repoObject to delete it.